pywr-schema/src/metric_sets/mod.rs

```diff
@@ -81,6 +81,46 @@ impl From<MetricAggregator> for pywr_core::recorders::Aggregator {
     }
 }
 
+#[derive(Deserialize, Serialize, Clone, JsonSchema, Default)]
+struct MetricSetFilters {
+    #[serde(default)]
+    all_nodes: bool,
+    #[serde(default)]
+    all_parameters: bool,
+}
+
+#[cfg(feature = "core")]
+impl MetricSetFilters {
+    fn create_metrics(&self, args: &LoadArgs) -> Option<Vec<Metric>> {
+        use crate::metric::{NodeReference, ParameterReference};
+
+        if !self.all_nodes && !self.all_parameters {
+            return None;
+        }
+
+        let mut metrics = vec![];
+
+        if self.all_nodes {
+            for node in args.schema.nodes.iter() {
+                metrics.push(Metric::Node(NodeReference::new(node.name().to_string(), None)));
+            }
+        }
+
+        if self.all_parameters {
+            if let Some(parameters) = args.schema.parameters.as_ref() {
+                for parameter in parameters.iter() {
+                    metrics.push(Metric::Parameter(ParameterReference::new(
+                        parameter.name().to_string(),
+                        None,
```

Comment:
I think this might not work if the parameter is a multi-valued one.

Reply:
It doesn't. I've added a temp fix which means that a Python parameter with a dict return value is skipped by the metric set filter. I think this is currently the only multi-valued parameter?

A better solution might be to create a separate name-spaced output metric for each return value of a multi-valued parameter. This might require the Python parameter to be updated so that these values are defined upfront by the user.

Reply:
It probably makes sense that each parameter struct in the schema has a method by which it can return a `ParameterReference` to itself.

Reply:
Should the nodes all have a similar method as well?

Reply:
Yes, probably.